### PR TITLE
35coreos-live/coreos-livepxe-rootfs: add in connectivity check before downloading

### PR DESCRIPTION
**Pull request by dustymabe**
This should help us get around some race conditions on startup
where we've seen curl exit with "No route to host" errors. This
most likely happens because the Networking in the kernel is still
being brought up (seen more in complex networking scenarios) and
curl hits that error the first time it tries. Since "No route to
host" isn't considered retryable, curl exits.

Instead let's just verify we can get to the remote at all in an
initial `curl --head` call. In this one we'll use --retry-all-errors
so that we will retry all errors. Once the connectivity to the remote
is verified then it should be safe to start downloading.

### CODE REVIEW

**Comment by dustymabe** [comment]
cc @bgilbert

**Review by cgwalters — DISMISSED** [review]

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh [hunk 1]
```diff
@@ -30,6 +30,18 @@ elif [[ -n "${rootfs_url}" ]]; then
...
         exit 1
     fi
+
+    # First, reach out to the server to verify connectivity before
+    # trying to download and pipe content through other programs.
+    # Doing this allows us to retry all errors (including transient
+    # "no route to host" errors during startup), without using the
+    # --retry-all-errors, which is problematic (see curl man page)
+    # when piping the output.
+    if ! curl --head --silent --show-error --insecure --location --retry 5 --retry-all-errors "${rootfs_url}"; then
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: Would be good to be able to share the common args between the two `curl` invocations. Maybe simply:

```
curl_args="--silent --show-error --insecure --location --retry 5"
```

?

**Review by jlebon — DISMISSED** [review]

File: overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh (hunk 1 above)
Comment (dustymabe) [reply overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: done

Comment (dustymabe) [reply overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh]: done

**Comment by dustymabe** [comment]
This looks good in local tests.

**Review by jlebon — APPROVED** [review]
LGTM! ISTM though like we should just match Ignition and retry forever instead of limiting to just 5 retries. Anyway, something we can discuss and do as a follow-up.

**Review by bgilbert — COMMENTED** [review]
LGTM

**Comment by miabbott** [comment]
@dustymabe I think this broke the `pxe-install` scenario on RHCOS; the `--retry-all-errors` option doesn't appear to be available in the RHEL version of `curl`

See this internal MR - https://gitlab.cee.redhat.com/coreos/redhat-coreos/-/merge_requests/1252

**Comment by jlebon** [comment]
Follow-up in https://github.com/coreos/fedora-coreos-config/pull/964.

**Pull request merged**